docker/test/performance-comparison/compare.sh

```diff
@@ -22,7 +22,7 @@ function configure
     echo all killed
 
     set -m # Spawn temporary in its own process groups
-    left/clickhouse-server --config-file=left/config/config.xml -- --path db0 &> setup-server-log.log &
+    left/clickhouse-server --config-file=left/config/config.xml -- --path db0 --user_files_path db0/user_files &> setup-server-log.log &
```

Comment:
Looks like it defaults to the same path: https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/Server.cpp#L455

What tests do you want to fix? For `insert_values_with_expressions`, apparently some files have to be downloaded, I'm not sure which files and where to get them.

Reply:
Found it in the old tests:
https://github.com/ClickHouse/ClickHouse/blob/ee5cf9f7bd9ba262146f088b4aa9b33acf1302a3/docker/test/performance/s3downloader#L17

In the new ones, it should be added here:
https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/performance-comparison/download.sh#L23

And it is already there... :thinking:

Reply:
> Looks like it defaults to the same path: https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/Server.cpp#L455

But apparently it not, because the error message says `File /workspace/user_files/test_all_expr_matches.values doesn't exist.`, when it should be `/workspace/right/db/user_files/test_all_expr_matches.values`.

Reply:
Ahh so that's where it is
https://github.com/ClickHouse/ClickHouse/blob/master/programs/server/config.d/path.xml#L4

So we should add something like `<user_files_path remove="remove"/>` here:
https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/performance-comparison/config/config.d/perf-comparison-tweaks-config.xml

Not sure about the relative priorities of these files...

Reply:
Right, all required files are downloaded, but paths in configs are wrong. Default `config.xml` sets path to `/var/lib/clickhouse/`, then `config.d/path.xml` overrides it to `./` and then command line argument overrides it to the correct value. Since `user_files_path` still specified in config, it is not determined as `path + "user_files/"` and without additional command line argument server searches files in wrong directory.

Reply:
Maybe we should add it to `perf-comparison-tweaks-config.xml` instead? Having all tweaks in one file makes them easier to find.
I'm not sure why I even added the path as a command line parameter -- out of habit perhaps, I run the server this way locally.

Reply:
I tried to add `<user_files_path remove="remove"/>` in `perf-comparison-tweaks-config.xml`, but it doesn't work because Poco interprets it as empty `user_files_path` and expressions like `config().getString("user_files_path", path + "user_files/")` return empty string instead of default value. Maybe we should add all paths to this config, but replacing all of them with `sed` looks worse then command line argument.